lib/actions/purgeTable.js

```diff
@@ -0,0 +1,77 @@
+let tableName = null;
+let dynamodb = null;
+
+const findPrimaryKey = () => {
+  const params = {
+    TableName: this.tableName
+  }
+
+  return new Promise((resolve, reject) => {
+    this.dynamodb.describeTable(params).promise()
+      .then(tableDescription => {
+        resolve({
+          primaryKey: tableDescription.Table.KeySchema.filter((element) => element.KeyType === "HASH")[0].AttributeName
+        })
+      })
+  })
+}
+
+const deleteAllElements = (deleteRequest) => {
+  const deleteRequests = [];
+
+  for (item of deleteRequest.items) {
+    let params = {
+      TableName: this.tableName,
+      Key: {}
+    }
+
+    params.Key[deleteRequest.primaryKey] = item[deleteRequest.primaryKey];
+
+    deleteRequests.push(this.dynamodb.deleteItem(params).promise());
+  }
+
+  return Promise.all(deleteRequests)
+}
+
+const findAllElements = (deleteRequest) => {
+  const params = {
+    TableName: this.tableName
+  }
+
+  let items = [];
+
+  return new Promise((resolve, reject) => {
+    this.dynamodb.scan(params).promise()
+      .then(results => {
+        items = results.Items;
+        deleteRequest.items = items
+        resolve(deleteRequest)
```

Comment:
Same issue as above for this regarding using unnecessary Promise.

Also, scan is limited to 200 items I believe, so this is not robust enough.

Maybe we should consider deleting the table instead?

Reply:
Actually it does seem to try to delete all items.
(But in my case I'm almost always getting some error if deleting a lot of item (~1000))